consoles/sshVirtsh.pm

```diff
@@ -571,7 +571,7 @@ __END"
         $self->get_cmd_output('echo bios.bootDelay = \"10000\" >> /vmfs/volumes/datastore1/openQA/' . $self->name . '.vmx', {domain => 'sshVMwareServer'});
     }
 
-    $ret = $self->run_cmd("virsh $remote_vmm start " . $self->name);
+    $ret = $self->run_cmd("virsh $remote_vmm start " . $self->name . ' 2> >(tee /tmp/os-autoinst-' . $self->name . '-stderr.log >&2)');
```

Comment:
you are saving logs to /tmp/ which is not controlled by the worker so potentially log files could pile up but you write to the same file per worker instance, should be ok.

The redirect looks a bit unusual. Is there a good reason to use this and not e.g. `… 2>&1 | tee …`?

Reply:
> you are saving logs to /tmp/ which is not controlled by the worker so potentially log files could pile up but you write to the same file per worker instance, should be ok.

The idea is that the file is used a flag "need to restart libvirt". And a restart of the machine could also cleanup /tmp/.


> The redirect looks a bit unusual. Is there a good reason to use this and not e.g. `… 2>&1 | tee …`?

Yes, I want that the STDERR is still sent as STDERR, with your proposal, it will go to STDOUT. With my redirect, someone that executes isotovideo manually can still benefit from filtering STDERR only.

Reply:
...and I only want STDERR on that file, not STDOUT

Reply:
...and only for the last executed command. If it fails, it will still until libvirt is restarted.
and, if libvirt/machine is restarted, the next command will empty that file, to the flag should be interpreted as "KVM memory allocation ok"

Reply:
> > you are saving logs to /tmp/ which is not controlled by the worker so potentially log files could pile up but you write to the same file per worker instance, should be ok.
> 
> The idea is that the file is used a flag "need to restart libvirt". And a restart of the machine could also cleanup /tmp/.

But when you restart the libvirt service this won't magically restart a machine or cleanup /tmp, what do you mean by that?

Reply:
no, but the operator can,

- restart libvirt and empty/remove the file
- restart libvirt and restart the failing job (it will empty the file)
- restart the machine (and if tmp is configure to be empty, ok, if not, manual remove or manual job restart)

Reply:
But, this is only the first step, my long-term goal is to automate it, so libvirt is automatically restarted (with proper cleanup) when no isotovideo is running (not possible to break a running job), etc.

Reply:
sure, makes sense